src/snapshot_middleware/project.rs

```diff
@@ -603,6 +576,29 @@ fn project_node_should_reserialize(
         }
     }
 
+    let db = rbx_reflection_database::get();
+    if let Some(class_descriptor) = db.classes.get(instance.class_name()) {
+        for (prop_name, inst_value) in instance.properties() {
+            if prop_name == "Attributes" {
+                continue;
+            }
+            let Some(default_value) = db.find_default_property(class_descriptor, prop_name) else {
+                continue;
+            };
+
+            if let Some(unresolved_node_value) = node_properties.get(prop_name) {
+                let node_value = unresolved_node_value
+                    .clone()
+                    .resolve(instance.class_name(), prop_name)?;
+                if variant_eq(&node_value, default_value) {
+                    return Ok(true);
+                }
+            } else if !variant_eq(inst_value, default_value) {
+                return Ok(true);
+            }
+        }
+    };
```

Comment:
If a class descriptor cannot be found, is "don't reserialize" the correct behavior?

Reply:
Same as with properties. If we don't know about a class, we probably just want to trust that it exists.